### ISSUE TITLE
`clippy` warns about use of `if let` in generated code

### ISSUE THREAD

**Issue by bruceg** [TedDriggs/darling#293]
When using `#[darling(default)]`, clippy warns that an `if let` can be simplified:
```
error: if let can be simplified with `.unwrap_or_default()`
   --> lib/vector-config-common/src/validation.rs:178:19
    |
178 |         #[darling(default, rename = "min")]
    |                   ^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_unwrap_or_default
```

The above lint is newly enabled in Rust 1.79.

I suspect the above is caused by the construct [here](https://github.com/TedDriggs/darling/blob/05113d1ba38ae24c255eb53f3c96e85702c2656f/core/src/codegen/field.rs#L227-L231), which indeed could be simplified with either `.unwrap_or(#expr)` or `.unwrap_or_default()` depending on the use case.

**Comment by maciektr** [comment]
There is a PR for this, but last activity is two weeks ago :)  https://github.com/TedDriggs/darling/pull/292

**Comment by TedDriggs** [comment]
This has been fixed by #296; release is imminent
